pkg/virtual/workspaces/registry/rest.go

```diff
@@ -678,28 +682,33 @@ func (s *REST) Delete(ctx context.Context, name string, deleteValidation rest.Va
 		Name:            internalName,
 		ResourceRequest: true,
 	}
-	if decision, _, err := authz.Authorize(ctx, deleteWorkspaceAttr); err != nil {
+	if decision, reason, err := authz.Authorize(ctx, deleteWorkspaceAttr); err != nil {
 		klog.Errorf("failed to authorize user %q to %q clusterworkspaces/workspace name %q in %s", userInfo.GetName(), "delete", internalName, orgClusterName)
 		return nil, false, kerrors.NewForbidden(tenancyv1beta1.Resource("workspaces"), "", fmt.Errorf("deletion in workspace %q is not allowed", orgClusterName))
 	} else if decision != authorizer.DecisionAllow {
+		klog.Errorf("user %q lacks (%s) clusterworkspaces/workspace %q permission for %q in %s: %s", userInfo.GetName(), decisions[decision], "delete", internalName, orgClusterName, reason)
+		return nil, false, kerrors.NewForbidden(tenancyv1beta1.Resource("workspaces"), internalName, fmt.Errorf("deletion in workspace %q is not allowed", orgClusterName))
```

Comment:
I wonder if this should be `name` instead of `internalName`?

Reply:
If you look up few lines, you'll see that there is still code in here around the internal/pretty name:

https://github.com/kcp-dev/kcp/blob/78dbe8e2d3b3ab897f67330918e161dba4dc798c/pkg/virtual/workspaces/registry/rest.go#L656-L667

So I think we are getting rid of that, but that would likely be a separate PR to clean it up.

Reply:
Looked at this some more.  The `NotFound` error was returned to the user, so it should use `name`.  the `klog.Errorf` is logging (not to the user obviously) about the `clusterworkspace`, which should reference the `internalName`.